handler-proxy/src/main/java/io/netty/handler/proxy/ProxyHandler.java

```diff
@@ -287,7 +287,7 @@ private void setConnectSuccess() {
             connectTimeoutFuture.cancel(false);
         }
 
-        if (connectPromise.trySuccess(ctx.channel())) {
+        if (!connectPromise.isDone()) {
             boolean removedCodec = true;
 
             removedCodec &= safeRemoveEncoder();
```

Comment:
consider using boolean operation while we are here (instead of bitwise operations)